### PR TITLE
refactor(fake_test_node): prefix package and namespace with autoware

### PR DESCRIPTION
**Pull request by esteve**
## Description

prefix package and namespace with `autoware`

## Related links

**Parent Issue:**

- https://github.com/autowarefoundation/autoware/issues/5077
- https://github.com/autowarefoundation/autoware/issues/4569

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by maxime-clem — APPROVED** [review]

File: common/autoware_fake_test_node/include/fake_test_node/fake_test_node.hpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #ifndef FAKE_TEST_NODE__FAKE_TEST_NODE_HPP_
 #define FAKE_TEST_NODE__FAKE_TEST_NODE_HPP_
 
-#include <fake_test_node/visibility_control.hpp>
+#include <autoware/fake_test_node/visibility_control.hpp>
```
Comment (maxime-clem) [inline common/autoware_fake_test_node/include/fake_test_node/fake_test_node.hpp:23]: This include fails because there is no `autoware` subfolder.

Comment (esteve) [reply common/autoware_fake_test_node/include/fake_test_node/fake_test_node.hpp]: Thanks for spotting this. Fixed in the latest changes.

**Comment by esteve** [comment]
@maxime-clem thanks for the review.

@xmfcx can you merge this? Thanks.

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/actions/runs/11815670194/job/32917329699?pr=9249#step:5:13854

> ```
> /__w/autoware.universe/autoware.universe/common/autoware_fake_test_node/test/test_fake_test_node.cpp:33:35: error: ‘tools’ in namespace ‘autoware’ does not name a type
>    33 | using FakeNodeFixture = autoware::tools::testing::FakeTestNode;
>       |                                   ^~~~~
> ```

CI has failed.

**Comment by esteve** [comment]
@xmfcx fixed in 6daea4b555802c3f4d4ac41079525fb6eb0aa048

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/actions/runs/11817499038/job/32923033722?pr=9249#step:5:20296

> ```
> CMake Error at /__w/autoware.universe/autoware.universe/install/ament_cmake_target_dependencies/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:77 (message):
>   ament_target_dependencies() the passed package name 'fake_test_node' was
>   not found before
> Call Stack (most recent call first):
>   CMakeLists.txt:36 (ament_target_dependencies)
> ```

It seems to have failed again.

**Comment by esteve** [comment]
@xmfcx sorry for all the back and forth, I've built this branch locally and the build passes. Hopefully CI passes as well.

**Pull request merged**